### PR TITLE
Add BloomFilter

### PR DESCRIPTION
**Pull request by ackintosh**
Hi, I've added a Bloom Filter implementation. Please review when you have time. 😃

### CODE REVIEW

File: bloomfilter/src/main/kotlin/org/kethereum/bloomfilter/BloomFilter.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                if (!bits.get(hashing(size, seed, value))) {
+                    return false
+                }
+            }
+        }
+        return true
+    }
+
+    private fun hashing(filterSize: Int, seed: Int, value: ByteArray) =
+            BigInteger(1, value.plus(seed.toByte()).sha256())
```
Comment (ackintosh) [inline bloomfilter/src/main/kotlin/org/kethereum/bloomfilter/BloomFilter.kt]: Using [sha256](https://github.com/komputing/KEthereum/tree/master/hashes/src/main/kotlin/org/kethereum/hashes) which is under `hashes` package for now but the hash algorithm may not appropriate. I'd like to try to implement more lightweight algorithm into the `hashes` package. 👀

File: bloomfilter/src/main/kotlin/org/kethereum/bloomfilter/BloomFilter.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    fun add(value: ByteArray) {
+        lock.write {
+            for (seed in 1..3) {
+                bits.set(hashing(size, seed, value))
+            }
+        }
+    }
+
+    fun mightContains(value: ByteArray): Boolean {
```
Comment (ligi) [inline bloomfilter/src/main/kotlin/org/kethereum/bloomfilter/BloomFilter.kt]: just a nit - I think mightContain would be a better function name

**Comment by ackintosh** [comment]
Thanks for the review! Fixed it. 😌

**Comment by ligi** [comment]
:+1:

**Pull request merged**